subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/junit/JUnitTestClassExecutor.java

```diff
@@ -116,6 +105,19 @@ private void runTestClass(String testClassName) throws ClassNotFoundException {
         runner.run(notifier);
     }
 
+    private void verifyCategoryClasses(Set<String> categories) {
+        if (categories == null) {
+            return;
+        }
+        for (String cls : categories) {
+            try {
+                applicationClassLoader.loadClass(cls);
```

Comment:
Is there any problem with the asymmetry between this and `loadClass()` above?  Here: you've just verified that the class exists within the `applicationClassLoader`, but that may not be the case when it gets loaded by the classloader used to check the category.

Reply:
Yes you're right that's an incorrect assumption. Now I use the same classloader to verify the class exists.